### ISSUE TITLE
FX particles systematically crashes if emitter_state_change is a parameter

### ISSUE THREAD

**Issue by Ragetto** [defold/defold#7469]
**Describe the bug (REQUIRED)**
Cf this topic:
https://forum.defold.com/t/fx-particles-make-my-game-crash-and-this-is-weird/72787

**To Reproduce (REQUIRED)**
Just launch a few spells in a row and it systematically crashes (cf the topic + the shared project)

**Expected behavior (REQUIRED)**
Should not crash :) I suppose

**Defold version (REQUIRED):**
  - Version 1.3.6

**Platforms (REQUIRED):**
 - Windows 11

**Minimal repro case project (OPTIONAL):**
Just shared this repo with @britzl and @JCash 
https://github.com/Ragetto/20230315_SHARED_FX_Particles_Systematic_Crash
This is the entire project, but I tried to skip most elements so the investigation is easier

Cf "screenshots" section for more details (walkthrough)

**Logs (OPTIONAL):**
Please provide relevant logs from engine, editor or build server. Learn where the logs are stored [here](https://defold.com/manuals/getting-help/#log-files).

**Workaround (OPTIONAL):**

![image](https://user-images.githubusercontent.com/69473434/225297304-40403203-7f3e-41a7-b050-a8f01afdb663.png)

```
function init(self)
	particlefx.play("#"..self.identifier, emitter_state_change) -- is causing the game crash
	--particlefx.play("#"..self.identifier) -- is used to avoid the crash (but is commented here)
end
```

```
local function emitter_state_change(self, id, emitter, state)
	if emitter == hash("emitter") and state == particlefx.EMITTER_STATE_POSTSPAWN then
		self.delete_object = true
		--print("[char_effect] order to delete the object!")
	end
end
```

As explained on the forum, when I remove the "emitter_state_change" function, it no longer crashes.

**Screenshots (OPTIONAL):**
1/ after launching the project, you just have to spawn ONE character by tapping this button:
![image](https://user-images.githubusercontent.com/69473434/225295558-62d0e9ca-4613-4a94-9978-cbc9c0a466e8.png)
2/ you add them to your formation:
![image](https://user-images.githubusercontent.com/69473434/225295679-594ca8f3-05f7-483f-bcf4-7a30a75f0af5.png)
![image](https://user-images.githubusercontent.com/69473434/225295924-074d7502-408a-4886-a0a1-841332c2df08.png)
3/ you go back to the home screen and then to the world map:
![image](https://user-images.githubusercontent.com/69473434/225296188-bc832c3e-c6fd-4400-ba80-e15fcc18011b.png)
4/ you go to the second region
![image](https://user-images.githubusercontent.com/69473434/225296858-47f5ff8e-c4e5-4245-b5f0-9d25adf7a6fd.png)
5/ you select the level 10
6/ you use the spell (particularly the lightning one)... the crash should occur pretty fast (cf video in the topic)

Don't worry, after the 1st time, you'll just have to click the campaign button (world map) and launch the level 10.

**Additional context (OPTIONAL):**

**Comment by Ragetto** [comment]
> are you sure that function `emitter_state_change` higher in your Lua script than a place you are using it (`init ` function) ?

Yes, sorry I put it below in this message but in the script it's above

I'll edit my initial post to make it clearer.

**Comment by AGulev** [comment]
I can't reproduce the issue anymore using the latest Defold 1.12.2
If you are still able to reproduce it please feel free to reopen it and provide updated example

## Issue summary

The report says `particlefx.play("#...", emitter_state_change)` could crash Defold 1.3.6 on Windows when spells repeatedly spawned particle FX, while removing the callback avoided the crash. The linked forum thread and comments point to the callback path, especially when the owning object/script is deleted as part of the effect lifecycle.

## Evidence

- Issue `#7469` is still open, but it only has the original report and one clarifying comment; there is no follow-up showing the bug still reproduces on current builds.
- The current callback path in `engine/gamesys/src/gamesys/scripts/script_particlefx.cpp:147` keeps `EmitterStateChangedScriptData` by reference, validates the callback, and logs/destroys it safely if `dmScript::SetupCallback()` fails instead of continuing into a bad Lua state.
- The particle FX component still copies callback/user data into the runtime instance in `engine/gamesys/src/gamesys/components/comp_particlefx.cpp:730`, and particle instances free that callback payload on teardown in `engine/particle/src/particle.cpp:67` and `engine/particle/src/particle.cpp:273`.
- The repository history contains a later fix, `802008a3c7` / PR `#7766` (`Particle effects crashfix`), whose description says it fixed a crash that happened when a script was removed after starting a particle FX with a state change callback.
- Related issue `#7142`, which shows the same crash signature in `dmGameSystem::EmitterStateChangedCallback`, was closed after maintainers stated that "the actual crash has been fixed".
- Current tests include a particle FX callback case in `engine/gamesys/src/gamesys/test/particlefx/particlefx_play.script:17` and `engine/gamesys/src/gamesys/test/test_gamesys.cpp:1464`, showing callback-based playback is now an expected supported path.

## Why this looks possibly fixed

The reported trigger in `#7469` matches the already-fixed callback crash family from `#7142`: particle FX state callbacks interacting badly with object/script destruction. The current tree contains the post-1.3.6 crashfix in the exact callback code path, plus a regression-style callback test, and I did not find any newer c
